.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: Continuous Integration
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 'releases/*'
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Print build information
+        run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.17"
+      - name: Check and unit test
+        run: make check unit-test
+      - name: Clone Temporal Docker compose
+        uses: actions/checkout@v2
+        with:
+          repository: temporalio/docker-compose
+          path: ./docker-compose
+      - name: Start Temporal server
+        run: |
+          docker-compose --project-directory ./docker-compose up &
+          go run ./.github/workflows/wait_for_server.go
```

Comment:
Do you need to wait for the `default` namespace too?

Reply:
No, the integration tests have logic to register the namespace if it doesn't exist and wait until it appears.